src/main/java/org/assertj/core/api/junit/jupiter/SoftAssertionsExtension.java

```diff
@@ -0,0 +1,119 @@
+ * {@code @Test}, {@code @RepeatedTest}, {@code @ParameterizedTest},
+ * {@code @TestFactory}, or {@code @TestTemplate}. This extension does not inject
+ * {@code SoftAssertions} or {@code BDDSoftAssertions} arguments into test
+ * constructors or class-level lifecycle methods.
+ *
+ * <h3>Scope</h3>
+ *
+ * <p>The scope of the {@code SoftAssertions} or {@code BDDSoftAssertions} instance
+ * managed by this extension begins when a parameter of type {@code SoftAssertions}
+ * or {@code BDDSoftAssertions} is resolved for a {@code @BeforeEach} method,
+ * test method, or {@code @AfterEach} method. The scope of the instance ends after
+ * all {@code @AfterEach} methods have been executed for the corresponding test.
+ * When the scope ends, {@code assertAll()} will be invoked on the instance to
+ * verify that no soft assertions failed.
+ *
+ * <h3>Example with {@code SoftAssertions}</h3>
+ *
+ * <pre class="code">
+ * {@literal @}ExtendWith(SoftAssertionsExtension.class)
+ * class ExampleTestCase {
+ * 
+ *    {@literal @}Test
+ *    void multipleFailures(SoftAssertions softly) {
+ *       softly.assertThat(2 * 3).isEqualTo(0);
+ *       softly.assertThat(Arrays.asList(1, 2)).containsOnly(1);
+ *       softly.assertThat(1 + 1).isEqualTo(2);
+ *    }
+ * }</pre>
+ *
+ * <h3>Example with {@code BDDSoftAssertions}</h3>
+ *
+ * <pre class="code">
+ * {@literal @}ExtendWith(SoftAssertionsExtension.class)
+ * class ExampleTestCase {
+ * 
+ *    {@literal @}Test
+ *    void multipleFailures(BDDSoftAssertions softly) {
+ *       softly.then(2 * 3).isEqualTo(0);
+ *       softly.then(Arrays.asList(1, 2)).containsOnly(1);
+ *       softly.then(1 + 1).isEqualTo(2);
+ *    }
+ * }</pre>
+ *
+ * @author Sam Brannen
+ * @since 3.13
+ */
+public class SoftAssertionsExtension implements ParameterResolver, AfterEachCallback {
+
+	@Override
+	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		boolean isTestLevel = extensionContext.getTestMethod().isPresent();
+		boolean isSoftAssertionsParameter = parameterContext.getParameter().getType() == SoftAssertions.class;
+		boolean isBddSoftAssertionsParameter = parameterContext.getParameter().getType() == BDDSoftAssertions.class;
+		return (isTestLevel && (isSoftAssertionsParameter || isBddSoftAssertionsParameter));
+	}
+
+	@Override
+	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		if (parameterContext.getParameter().getType() == SoftAssertions.class) {
+			return getStore(extensionContext).getOrComputeIfAbsent(SoftAssertions.class);
+		}
+		return getStore(extensionContext).getOrComputeIfAbsent(BDDSoftAssertions.class);
+	}
+
+	@Override
+	public void afterEach(ExtensionContext extensionContext) throws Exception {
+		SoftAssertions softAssertions = getStore(extensionContext).remove(SoftAssertions.class, SoftAssertions.class);
+		if (softAssertions != null) {
```

Comment:
More a matter of style but I'd rather use an `Optional` instead of `null` check, something like:
```
Optional.of(getStore(extensionContext).remove(SoftAssertions.class, SoftAssertions.class))
        .ifPresent(softAssertions::assertAll)
```
I think it should work but I haven't tried it

Reply:
If you slightly modify it as follows, it works.

```java
Optional.ofNullable(getStore(extensionContext).remove(SoftAssertions.class, SoftAssertions.class))
	.ifPresent(SoftAssertions::assertAll);
```

As a general rule of thumb, I would not create an `Optional` just to replace an `if` statement without using any other features of `Optional` such as mapping, filtering, etc.

But it's your code base. So feel free to change whatever I've done once/if you merge the work. 😇 

Having said that, I _may_ introduce a helper method that does that behind the scenes in order to create a mini DSL for those two blocks of code.

Reply:
Done in latest commit: 21f56775f63cfbbe4dbaccf2d06d4cca52e3537d